src/seed/runResult.ts

```diff
@@ -0,0 +1,35 @@
+
+// Strict typing of run results
+export enum RunResultType {
+  SUCCESS = 'SUCCESS',
+  FAILURE = 'FAILURE',
+}
+
+export type RunResult = SuccessResult | FailureResult
+export interface SuccessResult {
+  type: RunResultType.SUCCESS
+  warnings: Array<string>
+}
+
+export interface FailureResult {
+  type: RunResultType.FAILURE
+  warnings: Array<string>
+  errors: Array<Error>
+}
+
+export class Result {
+  public static makeSuccessResult(warnings?: Array<string>): SuccessResult {
```

Comment:
```suggestion
  public static Success(warnings?: Array<string>): SuccessResult {
```

maybe let short the names of these static methods?